### PR TITLE
[Pooling] Preserve BERT-family heads for raw logits

### PR DESCRIPTION
**Pull request by taneem-ibrahim**
## Purpose

`use_activation=False` incorrectly skipped BERT-family architectural poolers along with the final activation, producing incorrect raw logits. Make these poolers unconditional so the flag controls only the final activation.

## Reproducer

On upstream/main, the architectural head changes when only the final classification-activation flag is toggled:

```python
from types import SimpleNamespace
from unittest.mock import Mock

import torch

from vllm.config import ModelConfig, PoolerConfig
from vllm.model_executor.models.bert import BertPooler
from vllm.model_executor.models.modernbert import ModernBertPooler
from vllm.pooling_params import PoolingParams
from vllm.v1.pool.metadata import PoolingMetadata, PoolingStates


def metadata(use_activation: bool) -> PoolingMetadata:
    return PoolingMetadata(
        prompt_lens=torch.ones(1, dtype=torch.long),
        prompt_token_ids=None,
        prompt_token_ids_cpu=None,
        pooling_params=[
            PoolingParams(task="classify", use_activation=use_activation)
        ],
        pooling_states=[PoolingStates()],
    )


for name, pooler_cls, hf_config in (
    ("bert", BertPooler, SimpleNamespace(hidden_size=4)),
    (
        "modernbert",
        ModernBertPooler,
        SimpleNamespace(
            hidden_size=4,
            classifier_bias=True,
            classifier_pooling="cls",
            norm_eps=1e-5,
            norm_bias=True,
        ),
    ),
):
    model_config = Mock(spec=ModelConfig)
    model_config.pooler_config = PoolerConfig(seq_pooling_type="CLS")
    model_config.hf_config = hf_config
    model_config.head_dtype = torch.float32

    torch.manual_seed(0)
    pooler = pooler_cls(model_config)
    hidden_states = torch.tensor([[0.25, -0.5, 1.0, -1.5]])
    with_activation = pooler.head(hidden_states, metadata(True))
    without_activation = pooler.head(hidden_states, metadata(False))
    print(f"{name}: {torch.equal(with_activation, without_activation)}")
```

Run with:

```bash
.venv/bin/python /tmp/repro.py
```

Upstream/main prints `False` for both architectures. This branch prints `True` because the architecture-specific projection is unconditional; the final classifier activation remains controlled by `use_activation` in `ClassifierPoolerHead`.

## Output on main / on branch

The following is verbatim output from the B300 E2E result formatter. The validator used MRV2, `runner="pooling"`, seed 17, float16, two unequal-length prompts, `rtol=1e-2`, and `atol=1e-4`. It checked Hugging Face raw and activated parity, shape, ordering, finiteness, mixed per-request activation, post-mixed contamination, and checkpoint head-weight identity.

### upstream/main (`a8d1aa9c99`)

```text
upstream/main@a8d1aa9c99
cross-encoder/ms-marco-TinyBERT-L-2-v2: raw_allclose=False raw_max_abs=17.002912521362305 raw_max_rel=1.7883095741271973 activated_allclose=True default_allclose=True mixed_isolation=True head_weights_loaded=True
Alibaba-NLP/gte-reranker-modernbert-base: raw_allclose=False raw_max_abs=25.676973342895508 raw_max_rel=13.462642669677734 activated_allclose=True default_allclose=True mixed_isolation=True head_weights_loaded=True
Alibaba-NLP/gte-multilingual-reranker-base: raw_allclose=False raw_max_abs=0.04290294647216797 raw_max_rel=0.04708747938275337 activated_allclose=True default_allclose=True mixed_isolation=True head_weights_loaded=True
FAILURES_JSON=[]
OVERALL_STATUS=PASS
[run_logged] exit_code=0 duration_seconds=51.789
```

`OVERALL_STATUS=PASS` means the expected main behavior was observed: activated parity passed and raw parity failed for all three model families.

### This branch (`4c9e86ec21`)

```text
branch@4c9e86ec21
cross-encoder/ms-marco-TinyBERT-L-2-v2: raw_allclose=True raw_max_abs=0.004235267639160156 raw_max_rel=0.00044545132550410926 activated_allclose=True default_allclose=True mixed_isolation=True head_weights_loaded=True
Alibaba-NLP/gte-reranker-modernbert-base: raw_allclose=True raw_max_abs=0.0017800331115722656 raw_max_rel=0.0011471075704321265 activated_allclose=True default_allclose=True mixed_isolation=True head_weights_loaded=True
Alibaba-NLP/gte-multilingual-reranker-base: raw_allclose=True raw_max_abs=0.0008299052715301514 raw_max_rel=0.003989779390394688 activated_allclose=True default_allclose=True mixed_isolation=True head_weights_loaded=True
FAILURES_JSON=[]
OVERALL_STATUS=PASS
[run_logged] exit_code=0 duration_seconds=43.823
```

For every checkpoint, the BERT/ModernBERT architectural-head tensors and final classifier tensors matched the corresponding Hugging Face checkpoint tensors exactly after dtype normalization, including matching SHA-256 prefixes and zero maximum weight error.

## Test Plan

Permanent MRV2 classification regression tests:

```bash
CUDA_VISIBLE_DEVICES=1 VLLM_USE_V2_MODEL_RUNNER=1 \
  .venv/bin/python -m pytest \
  tests/models/language/pooling/test_classification.py::test_classifier_architecture_pooler_is_unconditional \
  tests/models/language/pooling/test_classification.py::test_bert_model_runner_v2 \
  -vv -s
```

```text
3 passed, 14 warnings in 45.15s
```

Verified both required GTE reranker parameters were selected:

```bash
CUDA_VISIBLE_DEVICES=1 VLLM_USE_V2_MODEL_RUNNER=1 \
  .venv/bin/python -m pytest \
  tests/models/language/pooling_mteb_test/test_gte.py \
  -k test_rerank_models_mteb --collect-only -vv
```

```text
collected 28 items / 26 deselected / 2 selected
Running 2 items in this shard: tests/models/language/pooling_mteb_test/test_gte.py::test_rerank_models_mteb[model_info0], tests/models/language/pooling_mteb_test/test_gte.py::test_rerank_models_mteb[model_info1]
```

Existing GTE reranker MTEB evaluations:

```bash
CUDA_VISIBLE_DEVICES=1 VLLM_USE_V2_MODEL_RUNNER=1 \
  .venv/bin/python -m pytest \
  tests/models/language/pooling_mteb_test/test_gte.py \
  -k test_rerank_models_mteb -vv -s
```

```text
Model: Alibaba-NLP/gte-reranker-modernbert-base
VLLM: dtype:torch.float16 head_dtype:torch.float32 0.33515
SentenceTransformers: Constant 0.33386
Difference: -0.0012900000000000134
Model: Alibaba-NLP/gte-multilingual-reranker-base
VLLM: dtype:torch.float16 head_dtype:torch.float32 0.33037
SentenceTransformers: Constant 0.33062
Difference: 0.000250000000000028
2 passed, 26 deselected, 22 warnings in 106.91s
```

## AI assistance disclosure

OpenAI Codex (GPT-5) assisted with drafting the implementation.

### CODE REVIEW

File: vllm/model_executor/models/modernbert.py [hunk 1]
```diff
@@ -364,11 +363,11 @@ def __init__(self, model_config: ModelConfig):
             dtype=head_dtype,
         )
 
+        # Keep the model's head layers even when output activation is disabled.
         # Use lambdas so that weights are not registered under `self.head`
         self.head = EmbeddingPoolerHead(
```
Comment (yewentao256) [inline vllm/model_executor/models/modernbert.py]:
```suggestion
        # Use lambdas so that weights are not registered under `self.head`
        self.head = EmbeddingPoolerHead(
```

Comment (yewentao256) [reply vllm/model_executor/models/modernbert.py]: Comment seems redundant

**Review by yewentao256 — APPROVED** [review]
LGTM, thanks for the work!

File: vllm/model_executor/models/modernbert.py (hunk 1 above)
Comment (taneem-ibrahim) [reply vllm/model_executor/models/modernbert.py]: The reason I added the comment is  to clarify that these are required architectural layers, not the optional output activation. I thought `use_activation` name could make the new behavior look contradictory.
